flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java

```diff
@@ -85,6 +92,8 @@ public class Buckets<IN, BucketID> {
 
     private final BucketStateSerializer<BucketID> bucketStateSerializer;
 
+    private final ExecutorService snapshotActiveBucketsThreadPool;
```

Comment:
some comments:
- I am not convinced that this is covered by existing tests as there is a new thread pool that has been introduced. 
- Also I would like to see a test showing the expected behaviour when the CompletionException is thrown. I am unsure what we are looking to do when there is a  CompletionException is thrown. Does everything get cleaned up when this exception is thrown?
- the title talks of close buckets but the code is in snapshotActiveBuckets. It would be easier to understand if I could see an obvious connection between close buckets and snapshotActiveBuckets.
- I see the Jira talks about introducing an option for this - but the code has not done this - it has changed it from synchronous to asynchronous. what is the thinking here?